### DISCUSSION TITLE
Is there any way to use a runtime `pydantic.BaseModel` type for validation?

### DISCUSSION THREAD

**Discussion by alvarobartt** [pydantic/pydantic#5723] [Question]
Hi everyone!

Basically I'm trying to make sure that some dictionaries match a certain schema, but one of the fields of the schema is another dictionary. So what I want to do is to infer the schema for the nested dictionary once, and then automatically use it for validating any dictionary.

Lets assume I've a schema `A` as follows:

```python
from pydantic import BaseModel

class A(BaseModel):
  a: Dict[str, Any]
  b: int
  c: float
```

And I want to constraint what's provided as `A.a`, since `Dict[str, Any]` is too open, so that the nested `pydantic.BaseModel` schema is also validated, but I won't know the schema to validate until I get the first entry, which is going to determine the validation schema for the rest of the entries.

I know I can use `pydantic.create_model` to generate a dynamic `pydantic.BaseModel` for any given dictionary in this case, but how can I generate it and then "assign" it to `A.a` so that the type-validation is updated accordingly?

Assuming I have a Python dictionary as:

```python
entries = [{"a": {"d": [1, 2, 3]}, "b": 1, "c": 1.0}, {...}]
```

I want to generate a `pydantic.BaseModel` for `A.a`, which I can get as:

```python
from pydantic import create_model

unknown_schema = A(entries[0]).a  # Is `{"d": [1, 2, 3]}`
NewSchema = create_model("NewSchema", **{key: (type(value), ...) for key, value in unknown_schema.items()})
```

So that I use `NewSchema` as the type-validation of `A.a` instead of the default `Dict[str, Any]`.

I know that this implies a core conflict with the static type validation, so I thought of using a `TypeVar` named `UnknownSchema` that bounds to a `pydantic.BaseModel` and define the type of `A.a` as `Union[UnknownSchema, Dict[str, Any]]`, but I think that's not correct either...

So any guess on what can I do to define a dynamic type-validation for a nested `pydantic.BaseModel`? I can just avoid using `pydantic.BaseModel` in case the validation I'm looking for doesn't make any sense within the scope of `pydantic` itself, or even on type-validation; but I wanted to get some advice first, in case anyone knows! Thanks 🤗

**Comment by chrisK824** [comment]
Hmm, if you can create a model from the first entry, why not create the outter model as well on the fly, using the newly created at the time inner schema?

**Reply by alvarobartt** [reply]
Yes, one could do so, the thing is that I want to include the parent-schema as a type-hint too e.g. 

```python
from pydantic import BaseModel

class A(BaseModel):
  a: Dict[str, Any]
  b: int
  c: float
```

to be the type-hint of the function e.g.

```python
def do_something() -> A:
  ...
  return A(**{...})
```

so that in the return-statement the validation is done over e.g.

```python
from pydantic import BaseModel

class A(BaseModel):
  a: NewSchema
  b: int
  c: float
```

when `NewSchema` can be e.g:

```python
class NewSchema(BaseModel):
  d: List[int]
```

I don't know if it's clear, but it's not just for the type-hint but also for validating the whole thing before returning it.

**Reply by chrisK824** [reply]
Sorry for the late answer @alvarobartt . It seems that you need some other logic for what you need. If constructing the schema on the fly isn't enough then sure the approach should be different here.

**Reply by alvarobartt** [reply]
No problem @chrisK824! I'm now constructing the schema on the fly, and then validating it manually which is enough for what I need right now, but thanks for the response! Much appreciated 🤗
